app/policies/case/sar/internal_review_policy.rb

```diff
@@ -20,5 +24,11 @@ def respond_and_close?
     clear_failed_checks
     user.managing_teams.include?(self.case.managing_team)
   end
+
+  private
+
+  def user_is_an_approver_on_case?
+    self.case.approvers.include?(user)
+  end
```

Comment:
if you change the `edit?` to the one suggested above ( if it works , I didn't try it although I think it should work 😳 ), this function is not needed any more. Actually this one is duplicated with one in the `Workflows::Predicates#user_is_approver_on_case `